### DISCUSSION TITLE
When to publish new npm packages between major releases?

### DISCUSSION THREAD

**Discussion by sideshowbarker** [validator/validator#1185] [Infrastructure]
We should decide when (how often) to publish new NPM packages between releases.

cc @nschonni 

Currently the only NPM releases that happen with certainty are those for major releases of vnu itself. And the most-recent major release of vnu was the 20.6.30 release, more than one year ago.

In the mean time, for every push to main, we publish snapshot `latest`-tagged releases to https://github.com/validator/validator/releases/tag/latest, and also for every push to main and every new tag, we publish a new Docker image to https://github.com/validator/validator/pkgs/container/validator.

But for npm publishing, it seems clear we don’t want to publish a new npm package for every push to master. (And in fact, with our version-numbering scheme, as can only publish to a new npm once per day — if we push to main more than once a day, the associated npm package wouldn’t get updated.)

So what I’ve been doing in practice is, I’ve just been publishing new npm packages on-demand, when somebody asks me to publish a new one. But it would be good have some strategy other than just “when somebody asks for a new package”.

Orthogonal to the question of when we should publish new npm packages is the question of whether we should automate the publication (in our CI/GitHub Actions).

Given that it takes literally only a few seconds to run `./checker.py npm-release` from the command line locally to publish a new npm, and given that we don’t want/need to trigger npm releases based on any particular repo events, it’s not clear to me what value there’d be to adding CI automation for it.

**Comment by nschonni** [comment]
2 approaches come to mind:
1. When you tag locally and push a new tag, create a GitHub release + publish to NPM
2. Set a monthly cron job, that would create an new NPM release on the first of the month, assuming something has changed

**Reply by takenspc** [reply]
We deploy new vnu to an internal instance on a regular monthly basis. We would appropriate if you could:

> 2. Set a monthly cron job, that would create an new NPM release on the first of the month, assuming something has changed

**Comment by j9t** [comment]
@sideshowbarker, what’s your position on this these days? Would publishing on minor releases be useful, as a compromise between not rolling out every patch, but also not to wait just for major updates?

**Comment by sideshowbarker** [comment]
Thanks for the bump.

I guess given that the project is no longer making any distinction between major and minor releases, this question is now moot.

So I’m going ahead and closing it.
